packages/tpetra/core/test/MatrixMatrix/MatrixMatrix_UnitTests.cpp

```diff
@@ -687,9 +688,18 @@ mult_test_results multiply_reuse_test(
   computedC2->resumeFill();
   Tpetra::MatrixMatrix::Multiply(*As, AT, *Bs, BT, *computedC2, true/*call_FillCompleteOnResult*/);
 
+  // computedC1 = leftScaling * (op(A)*op(B)) * rightScaling
+  RCP<Matrix_t> computedC1 = rcp( new Matrix_t(map, computedC2->getGlobalMaxNumRowEntries()));
```

Comment:
I'm not sure how I feel about this.  Now we're tying these tests to a global quantity that could be quite large.  I guess we can always go back and tighten the bounds later.

Reply:
Understood.  This max value is not needed if Multiply fillCompletes, but this specific test is for the case without fillComplete.  So unless we do some sort of symbolic phase first to figure out the nonzero pattern, we have to use a max.    Indeed, this case is a bit artificial anyway, as we only have the max available because we've done the multiplication a different way already.  We could deprecate the non-fillCompleted version at some point.

Reply:
@kddevin ~ 2015, we talked about another option to do thread-parallel graph creation: Add a new graph class that bounds the total number of entries per _process_, but does not fix the number of entries per _row_.  The class would use `Kokkos::UnorderedMap` inside.  (That might have been before UnorderedMap existed, so it was just a theory.)  That could be a helpful future option for users who only want to build a graph once.  Tpetra already has a CooMatrix that could be a start for such a class.